knimin/static/js/platemapper.js

```diff
@@ -0,0 +1,506 @@
+    row = parseInt($(current).attr('pm-well-row'));
+    col = parseInt($(current).attr('pm-well-column'));
+    // Update indices
+    row = row + 1;
+    if (row === this.rows) {
+      row = 0;
+      col = col + 1;
+      if (col === this.cols) {
+        col = 0;
+      }
+    }
+    // Set the focus to the next input tag
+    this.wellInformation[row][col].inputTag.focus();
+  }
+}
+
+/**
+ *
+ * Change event handler on the well input
+ *
+ * @param {Elemnt} current The <input> element where the event has been trigered
+ * @param {Event} e The event object
+ *
+ **/
+PlateMap.prototype.changeWell = function (current, e) {
+  var sample, sampleInfo, row, col;
+  sample = $(current).val().trim();
+
+  sampleInfo = this.samples[sample];
+
+  // Clean any extra style that we added
+  $(current).removeClass('pm_sample_plated');
+
+  if (sampleInfo === undefined) {
+    // This sample is not recognized - mark the well as problematic
+    $(current).css({'background-color': 'red'});
+  } else {
+    // This sample belongs to a study - label the backgroun with the
+    // same color as the study
+    $(current).css({'background-color': sampleInfo.color});
+    // Check if it has been plated before
+    if (sampleInfo.plates.length > 0) {
+      $(current).addClass('pm_sample_plated');
+    }
+  }
+
+  this.updateWellCommentsArea();
+  // TODO: control the proceed to extraction button
+}
+
+/**
+ *
+ * Show event handler on the well comment modal
+ *
+ **/
+PlateMap.prototype.commentModalShow = function () {
+  var row = parseInt($('#comment-modal-btn').attr('pm-row'));
+  var col = parseInt($('#comment-modal-btn').attr('pm-col'));
+  // Magic number + 1 -> correct index because JavaScript arrays start at 0
+  var wellId = String.fromCharCode('A'.charCodeAt() + row) + (col + 1);
+  var wellInfo = this.wellInformation[row][col];
+  var sample = wellInfo.inputTag.val().trim();
+  if (sample.length === 0) {
+    sample = 'No sample plated';
+  }
+  var value = (wellInfo.comment !== undefined) ? wellInfo.comment : "";
+  $('#well-comment-textarea').val(value);
+  $('#exampleModalLabel').html('Adding comment to well ' + wellId + ' (Sample: <i>' + sample + '</i>)');
+}
+
+/**
+ *
+ * Click event handler on the save button form the comment modal
+ *
+ **/
+PlateMap.prototype.commentModalSave = function () {
+  var row = parseInt($('#comment-modal-btn').attr('pm-row'));
+  var col = parseInt($('#comment-modal-btn').attr('pm-col'));
+  var wellInfo = this.wellInformation[row][col];
+  wellInfo.comment = $('#well-comment-textarea').val();
+  wellInfo.inputTag.addClass('pm_well_commented');
+  this.updateWellCommentsArea();
+  $('#myModal').modal('hide');
+}
+
+
+/**
+ *
+ * Constructs the HTML elements of a well
+ *
+ * @param {int} row The row of the well
+ * @param {int} column The column of the well
+ *
+ * @return {jQuery.Object} The div representing a well
+ *
+ **/
+PlateMap.prototype.constructWell = function(row, column) {
+  var obj = this;
```

Comment:
I had seen `this` being referred to as `that` before, I suggest just using one uniformly.